packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -1168,6 +1153,66 @@ export class ThinEngine extends AbstractEngine {
         this._bindUnboundFramebuffer(null);
     }
 
+    /**
+     * Resolves the MSAA textures of the render target into their non-MSAA version
+     * Note that if "texture" is not a MSAA render target, no resolve is performed but mipmaps will still be generated (except if disableGenerateMipMaps is true or if a texture has generateMipMaps equals to false).
+     * @param texture  The render target texture containing the MSAA textures to resolve
+     * @param disableGenerateMipMaps Defines a boolean indicating that mipmaps must not be generated (default: false)
+     */
+    public resolveFramebuffer(texture: RenderTargetWrapper, disableGenerateMipMaps = false): void {
+        const rtWrapper = texture as WebGLRenderTargetWrapper;
+        const gl = this._gl;
+
+        if (rtWrapper._MSAAFramebuffer && !rtWrapper.disableAutomaticMSAAResolve) {

```

Comment:
Thinking more about it, I do not think the flag should be here but only in the place calling automatically.

Reply:
Done.